src/widget.c

```diff
@@ -54,10 +54,18 @@ void _twin_widget_paint_shape(twin_widget_t *widget,
 {
     twin_pixmap_t *pixmap = widget->window->pixmap;
 
-    if (shape == TwinShapeRectangle)
+    if (shape == TwinShapeRectangle) {
+#if defined(CONFIG_PIXMAN)
+        pixman_color_t color;
+        twin_argb32_to_pixman_color(widget->background, &color);
+        pixman_fill_rect(pixmap, &color, pixmap->clip.left, pixmap->clip.top,
+                         right, bottom);
+        twin_pixmap_damage(pixmap, left, top, right, bottom);
+#else
         twin_fill(pixmap, widget->background, TWIN_SOURCE, left, top, right,
```

Comment:
I think you could rewrite `twin_fill` as following to hide the detailed implementation
```c
void twin_fill(...)
{
#if defined(CONFIG_PIXMAN)
...pixman library implementaion
#else
...original implementation
#endif
}
```